### ISSUE TITLE
log output of `command_id` processes to file

### ISSUE THREAD

**Issue by nalbion** [Pythagora-io/gpt-pilot#236]
When `cli.execute_command()` starts a process with `command_id` it will be left running in the background and we'll continue with the conversation after `timeout` or `success_message`.  Sometimes the LLM will ask you to send a request using the browser or Postman etc and "check the terminal output" but we don't have access to that anymore.

We want to be able to:
- [ ] send the initial stdout/err to the LLM, 
- [ ] ...but it would also be handy to log ongoing output to a log file - `.gpt-pilot/{command_id}.log`. (see https://github.com/Pythagora-io/gpt-pilot/blob/main/pilot/utils/dot_gpt_pilot.py)
 - [ ] Terminating an existing process before restarting with the same ID should reset the log file.

- [ ] Ideally, we would provide a `function` or `tool` for the LLM to read the output for itself. (#239)

- [ ] We could refactor a lot of the code in `cli.execute_command()` into 2 or 3 implementations of a `Process` class.

# Current implementation:
```python
    q_stderr = queue.Queue()
    q = queue.Queue()
    process = run_command(command, project.root_path, q, q_stderr)

    if command_id is not None:
        terminate_named_process(command_id)
        # TODO: We want to be able to send the initial stdout/err to the LLM, but it would also be handy to log ongoing output to a log file, named after `command_id`. Terminating an existing process with the same ID should reset the log file
        running_processes[command_id] = (command, process.pid)

    output = ''
    stderr_output = ''
```

# Proposed implementation
```python
    process = ForegroundProcess() if command_id is None else BackgroundProcess(process_id)
    process.run(command, project.root_path)
...  
    process.check_if_exited()    # read from q or log file
...
    if timeout...
        process.on_timeout()     # log or raise TimeoutError
        break
...
    output, stderr_output = process.read_output()
```

**Comment by Umpire2018** [comment]
Assign to me please.

**Comment by sarahedwin442** [comment]
I want to use this golden medium to Appreciate Dr Kuda for making my dream come through. I was on the internet when I saw people posting and talking about Dr Kuda. On How he has help them in winning huge amounts in lottery ticket, My name is Sarah Edwin, this is my story on how I win $754.6 Million Dollars, After reading the article talking good about Dr Kuda I contacted Dr Kuda I told him to help me win the Powerball lottery, Then he respond to me by saying he will help me after working with Dr Kuda, and assured me that I will win the Powerball Lottery Draw, Which I believe. After working with Dr Kuda, he gave me a five digit number and ask me to go and play it which I did today, now i’m here shearing a testimony about this same man Dr Kuda Changed my life by helping me to win $754.6 Million Dollars all thanks to you Dr Kuda, I will keep talking about your good work, because you are a man of your word I hope people that really need this help will come across it for you to help them as well. If you want to win biggest in lottery contact Dr. Kuda today and be rest assured of winning lottery game, contact via email: drkudaspelltemple@gmail.com Or Call and Whatsapp Number +1 (346) 478-1991
